### PR TITLE
Rollup Rework And Improvement

### PR DESCRIPTION
**Pull request by nickreese**
## Testing

The testing suite is now windows compatible. This was a major major undertaking but having rewritten almost 200 tests to accommodate it, we're glad it is available.

## CSS: Inline or External File

There are now 2 css modes. 

1. One where all of the CSS will be written to a file and the proper includes will be added to the head of the html.
2. One where all CSS a component depends on will be added to the head. 

The 2nd seems easy but is far from trivial. To do this we trace all of the dependencies of a file, gather the css from those files, merge the source maps, and add it as a separate export to the CJS file that is generated. 

The "why" behind the 2nd option is to be able to create truly standalone SSR versions of Svelte components.

## Features:

* can now write css to an external file. Simply add `css: "file"` to your `elder.config.js`
* Allows for importing of css in svelte files.
* consolidates all elderjs output into `/public/_elderjs`
* Now properly gathers all dependent css for a component. 

## Fixes:

* file naming issues
* css not getting updated in dev mode
* css order is preserved.
* CSS not catching all dependencies.

## Breaking:

* Split components during rollup needed to go. the complexity was too much. 
* consolidates all elderjs output into `/public/_elderjs`

### CODE REVIEW

**Review by myrsmedenstorytel — APPROVED** [review]
This seems to work really well, both with the case of an external library with UI components and mixed together with some styles in internal templates and components. Such great functionality!

File: src/partialHydration/__tests__/prepareFindSvelteComponent.spec.ts [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+    });
+
+    it(`properly removes hashes from components in non-nested`, () => {
+      expect(
+        [
+          '/_elderjs/svelte/components/AutoComplete.123456.js',
+          '/_elderjs/svelte/components/Breadcrumbs.123456.js',
+          '/_elderjs/svelte/components/Citation.123456.js',
+          '/_elderjs/svelte/components/CityResults.123456.js',
+        ].map((c) => removeHash(c)),
```
Comment (halafi) [inline src/partialHydration/__tests__/prepareFindSvelteComponent.spec.ts:299]: can just do `.map(removeHash)`

File: src/rollup/__tests__/rollupPlugin.test.ts [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+const rollup = require('rollup');
+
+describe('#rollupPlugin', () => {
+  const cfs = fsExtra.copyFileSync;
+  const rds = fsExtra.readdirSync;
+  const eds = fsExtra.ensureDirSync;
+
+  // @ts-ignore
+  fsExtra.copyFileSync = jest.fn(cfs);
```
Comment (halafi) [inline src/rollup/__tests__/rollupPlugin.test.ts:19]: can be chained with mockImplementation

https://jestjs.io/docs/en/mock-function-api#mockfnmockimplementationfn

**Review by halafi — COMMENTED** [review]
found just one typo and have some suggestions

**Comment by sw-yx** [comment]
had no idea this was coming. super nice work!

**Pull request merged**